plugins/hydra_orion_sweeper/hydra_plugins/hydra_orion_sweeper/_impl.py

```diff
@@ -0,0 +1,347 @@
+
+        self.space = None
+        self.arguments = dict()
+
+        log.debug("Starting launcher")
+
+        self.launcher = Plugins.instance().instantiate_launcher(
+            hydra_context=hydra_context, task_function=task_function, config=config
+        )
+
+    def n_workers(self):
+        n = self.worker_config.n_workers
+        if n <= 0:
+            return os.cpu_count()
+
+        return n
+
+    def suggest_trials(self, count) -> List[Trial]:
+        """Suggest a bunch of trials to be dispatched to the workers"""
+        trials = []
+
+        for _ in range(count):
+            try:
+                trial = self.client.suggest(pool_size=count)
+                trials.append(trial)
+
+            # non critical errors
+            except WaitingForTrials:
+                break
+
+            except ReservationRaceCondition:
+                break
+
+            except CompletedExperiment:
+                break
+
+        return trials
+
+    def new_experiment(self, arguments) -> ExperimentClient:
+        """Initialize orion client from the config and the arguments"""
+
+        self.space_parser.add_from_overrides(arguments)
+        self.space, self.arguments = self.space_parser.space()
+
+        dict_config = OmegaConf.to_container(self.algo_config)
+        algo_type = dict_config.pop("type", "random")
+        algo_config = dict_config.pop("config", dict())
+
+        log.info("Orion Optimizer %s", self.algo_config)
+        log.info("with parametrization %s", self.space.configuration)
+
+        return create_experiment(
+            name=self.orion_config.name,
+            version=self.orion_config.version,
+            space=self.space,
+            algorithms={algo_type: algo_config},
+            strategy=None,
+            max_trials=self.worker_config.max_trials,
+            max_broken=self.worker_config.max_broken,
+            storage=self.storage_config,
+            branching=self.orion_config.branching,
+            max_idle_time=None,
+            heartbeat=None,
+            working_dir=None,
+            debug=self.orion_config.debug,
+            executor=None,
+        )
+
+    def sweep(self, arguments: List[str]) -> None:
+        """Execute the optimization process"""
+
+        assert self.config is not None
+        assert self.launcher is not None
+        assert self.job_idx is not None
+
+        log.debug("Starting new experiment")
+        self.client = self.new_experiment(arguments)
+
+        with clientctx(self.client):
+            try:
+                self.optimize(self.client)
+            except Exception as e:
+                self.release_all()
+                raise e
+
+    def release_all(self) -> None:
+        """Make sure not trials remain reserved"""
+        for trial in self.pending_trials:
+            try:
+                self.client.release(trial, status="interrupted")
+            except AlreadyReleased:
+                pass
+
```

Comment:
This could be made part of clientctx.

Reply:
wouldn't be best to move it to Orion own ExperimentClient ?